### PR TITLE
Add available for hire button

### LINKED ISSUE #15357: Settings: "Are you available for hire?" button
We need to add a button to our settings that allows campers to mark themselves as "available for hire". This won't change anything visible about their profile, but it will let us know that we can send that camper some career-related advice through email, and better track campers who get jobs.

Here's what this interface looks like on GitHub:

your_profile

Here's what it looks like on CodePen:

We may be able to turn it into a single button that toggles on and off like:

"No, I'm not available for hire" -> "Yes, I am available for hire"

### PR DESCRIPTION
**Pull request by padulam**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #15357

#### Description

This update adds an available for hire button to the user settings page. The default setting for this button is 'unavailable'. The user can toggle the button from available/unavailable, similar to the other buttons on the settings page.

### CODE REVIEW

File: server/services/user.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const publicUserProps = [
   'isGithubCool',
 
   'isLocked',
+  'isAvailable',
```
Comment (raisedadead) [inline server/services/user.js]: This key should be better named to `isAvailableForHire`

File: server/boot/user.js [hunk 2]
```diff
@@ -463,6 +463,7 @@ module.exports = function(app) {
           isGithubCool: true,
           isCheater: true,
           isLocked: true,
+          isAvailable: true,
```
Comment (raisedadead) [inline server/boot/user.js]: This key should be better named to `isAvailableForHire`

File: server/boot/settings.js [hunk 3]
```diff
@@ -82,6 +82,11 @@ export default function settingsController(app) {
   }
 
   api.post(
+    '/toggle-available-for-hire',
+    ifNoUser401,
+    toggleUserFlag('isAvailable')
```
Comment (raisedadead) [inline server/boot/settings.js]: This key should be better named to `isAvailableForHire`

File: common/models/user.json [hunk 4]
```diff
@@ -120,6 +120,11 @@
       "description": "Campers profile does not show challenges/certificates to the public",
       "default": false
     },
+    "isAvailable": {
```
Comment (raisedadead) [inline common/models/user.json]: This key should be better named to `isAvailableForHire`

File: common/app/routes/settings/redux/update-user-saga.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ import combineSagas from '../../../../utils/combine-sagas';
 
 const urlMap = {
   isLocked: 'lockdown',
+  isAvailable: 'available-for-hire',
```
Comment (raisedadead) [inline common/app/routes/settings/redux/update-user-saga.js]: This key should be better named to `isAvailableForHire`

File: common/app/routes/settings/components/Settings.jsx [hunk 6]
```diff
@@ -219,6 +227,22 @@ export class Settings extends React.Component {
...
+        <Row>
+          <Col
+            md={ 6 }
+            mdOffset={ 3 }
+            sm={ 8 }
+            smOffset={ 2 }
+            xs={ 12 }
+            >
+            <JobSettings
+              isAvailable={ isAvailable }
```
Comment (raisedadead) [inline common/app/routes/settings/components/Settings.jsx]: This key should be better named to `isAvailableForHire`

File: common/app/routes/settings/components/Job-Settings.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      </Col>
+      <Col xs={ 3 }>
+        <Button
+          block={ true }
+          bsSize='lg'
+          bsStyle='primary'
+          className={ className }
+          onClick={ toggle }
+          >
+          { isAvailable ? 'Available' : 'Unavailable' }
```
Comment (raisedadead) [inline common/app/routes/settings/components/Job-Settings.jsx]: This key should be better named to `isAvailableForHire`

File: common/app/routes/settings/components/Job-Settings.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React, { PropTypes } from 'react';
+import { Button, Row, Col } from 'react-bootstrap';
+import classnames from 'classnames';
+
+const propTypes = {
+  isAvailable: PropTypes.bool,
```
Comment (raisedadead) [inline common/app/routes/settings/components/Job-Settings.jsx]: This key should be better named to `isAvailableForHire`

**Review by raisedadead — CHANGES_REQUESTED** [review]
This key should be better named to `isAvailableForHire` for sake of understanding

**Comment by padulam** [comment]
@raisedadead I updated all instances of the isAvailable key to isAvailableForHire. I also updated the toggleIsAvailable function to toggleIsAvailableForHire. Thanks!

**Review by BerkeleyTrue — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
Awesome! Changes LGTM, I'll let @BerkeleyTrue merge this.

Thanks a lot for this pull request and great work @padulam!!

**Pull request merged**